src/value/value.c

```diff
@@ -47,9 +47,10 @@ static inline mempool_t *getPool() {
 ///////////////////////////////////////////////////////////////
 
 static inline int cmp_strings(const char *s1, const char *s2, size_t l1, size_t l2) {
-  int cmp = strncmp(s1, s2, MIN(l1, l2));
+  // Use memcmp instead of strncmp to correctly handle binary data with embedded NULLs
+  int cmp = memcmp(s1, s2, MIN(l1, l2));
```

Comment:
Seems OK - `memcmp` is also comparing bytes, similar to `strncmp` - only supporting embedded nulls.
When comparing null-terminated strings - will behave the same as before - and no extra comparisons will be done.
When comparing slices, which can contain embedded nulls, will behave as expected.